pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -87,6 +88,12 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return fmt.Errorf("intrusiondetection-controller failed to watch installer job: %v", err)
 	}
 
+	// Watch for changes to to primary resource LogCollector, to determine if syslog forwarding is
+	// turned on or off.
+	// Since this syslog forwarding is optional, we don't need to return an error if LogCollector
+	// is not found.
+	_ = c.Watch(&source.Kind{Type: &operatorv1.LogCollector{}}, &handler.EnqueueRequestForObject{})
```

Comment:
Do we need to handle the error returned ?

Reply:
See my [comment above](https://github.com/tigera/operator/pull/941/files#r504936180). 

> Of course, the other angle to this is even for a management or standalone cluster, syslog forwarding is optional. So if it cannot watch a LogCollector then perhaps it should proceed as if nothing bad happened regardless?

My main concern was that LogCollector was not critical like the other resources ID controller watches for. So I was thinking returning an error here was too severe. 

But I suppose no matter what kind of cluster this is in, if the user has Intrusion Detection running then they must also have a LogCollector defined. 

... I'll add an error here.